### DISCUSSION TITLE
adapt medspacy for Spanish!

### DISCUSSION THREAD

**Discussion by juandelahoz** [medspacy/medspacy#44] [Ideas] [answered]
Hi! I love that you're doing this.
I am interested in extending medspacy for Spanish!
Would you like to let me know where would be the best place to start?
Juan

**Comment by turbosheep** [comment, marked as answer]
Hi, since we had a question about this just yesterday for French [here](https://github.com/medspacy/medspacy/discussions/43), the answer will be similar. 

We would love to add multilingual support for medspacy. Right now, QuickUMLS should work if resources are generated with the Spanish UMLS. Other components like the sectionizer and context are rule-based, so creating parallel or similar rule sets in Spanish would be a good place to start. You can look at the current default rules in medspacy's [resources folder](https://github.com/medspacy/medspacy/tree/master/resources). Our rules are still a work in progress, but they show a range of the possibilities that might need to be translated.

If you run into any bugs related to medspacy components processing Spanish text or rules, feel free to open an issue here as well.

Ultimately, these Spanish rules would be used alongside one of [spacy's Spanish models](https://spacy.io/models/es) for tokenizing or dependency parsing as needed. 

If you start work on any Spanish additions to medspacy, we would be happy work towards incorporating them into the project or link to your work on the README and in documentation. 

EDIT: @abchapman93 mentioned to me just now that another way to help would be to create a simple jupyter notebook that may or may not use medspacy components, but give simple examples of what clinical text looks like in Spanish. Although the data we work on sometimes contains other languages and some people on our team are multilingual, we do not necessarily know what clinical text in Spanish or other languages looks like. I assume it is as different from spoken Spanish as English clinical text is from spoken English!

**Reply by juandelahoz** [reply]
Thank you so much for the quick reply and for pointing me in the right direction!
I've created spaCy components with NegEx and ConText rules for my own work in Spanish, and will try to figure out how to adapt them for your framework. Actually, spaCy's Spanish models are one of the main reasons why this is my tool of preference.

There is this publicly available corpus of annotated clinical text in Spanish called [NUBes](https://github.com/Vicomtech/NUBes-negation-uncertainty-biomedical-corpus) that we are using to train negation algorithms. (Same, very different from spoken Spanish).

I'll be in touch with an update soon.

**Reply by abchapman93** [reply]
Oh very cool. This is very interesting, thanks for sharing! Please let us know if we can help adapt your resources to medspaCy, and if you'd be willing to share any Spanish resources you develop we'd love to include them in medspaCy! 

In case it's useful, #43 shows an example of converting another implementation of French ConText rules to medspaCy, so perhaps you'll be able to do something similar with your existing rules.

**Comment by abchapman93** [comment]
In addition to writing your own rules, some previous work has adapted other ConText implementations to Spanish. These weren't medspaCy, but were related algorithms that could possibly be put into medspaCy. Here's a paper (which I unfortunately can't access) that implements NegEx, an earlier version of ConText, into Spanish: https://link.springer.com/chapter/10.1007/978-3-319-09891-3_34

Since there's some interest around non-English medspaCy, I'll do some digging to see if I can find any openly available resources for adaptations of ConText and NegEx that might be useful.

**Reply by juandelahoz** [reply]
Thank you very much for doing some of the research! I'd be happy to see what other resources are freely available in Spanish.

**Comment by burgersmoke** [comment]
Good news on non-English work on medspacy.  I'm currently working on breaking up medspacy resources by language.   In doing this, English still works smoothly and I'm making great progress in French (my second language).

Next up is Spanish (my 3rd language).  Does anyone here have additional resources on Spanish language rules either for ConText or for NegEx?  @juandelahoz  or others?

If not, I'm thinking about trying to transform repos like this one into ConText rules:

https://github.com/PlanTL-GOB-ES/NegEx-MES

Can anyone think of any better resources?  @juandelahoz  I think you may have mentioned you already had some ConText rules in Spanish.  Is that right?  If so, could you share them with me?

**Comment by burgersmoke** [comment]
@juandelahoz  Do you happen to have any section rules that you or others might have developed for Spanish?  If so, could you send them to me?
